packages/scanner/test/cli/watchScan.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { Watcher, WatchScanOptions } from '../../src/cli/scan/watchScan';
+import { ScanResults} from '../../src/report/scanResults';
+import Configuration from '../../src/configuration/types/configuration';
+import { withStubbedTelemetry } from '../helper';
+
+describe('watchScan', () => {
+  withStubbedTelemetry();
+  const options: WatchScanOptions = {
+    appId: "appId",
+    appmapDir: 'tmp/dummy',
+    configFile: 'dummy_config',
+  };
+  const blankScanResults = new ScanResults(
+        {
+          checks: []
+        },
+        {},
+        [],
+        [],
+  );
+  
+  it('does not crash if there are no telemetry events', async () => {
+    const watcher: Watcher = new Watcher(options);
+    clearTimeout(watcher.interval); // disable the timer
+
+    watcher.sendScanResultsTelemetryInABatch();
+    expect(watcher.batchedTelemetry.length).toBe(0);
+  });
+
+  it('does not send telemetry events if the period has not elapsed', async () => {
+    const watcher: Watcher = new Watcher(options);
+    clearTimeout(watcher.interval); // disable the timer
+    
+    watcher.batchedTelemetry.push({
+      scanResults: blankScanResults,
+      msElapsed: 10,
+      msInserted: Date.now(),
+    });
+    watcher.sendScanResultsTelemetryInABatch();
+    expect(watcher.batchedTelemetry.length).toBe(1); // wasn't sent
+  });
+
+  it('sends telemetry events if the period elapsed', async () => {
+    const watcher: Watcher = new Watcher(options);
+    clearTimeout(watcher.interval); // disable the timer
```

Comment:
I see now we're actually calling `unref` on the timer elsewhere so this `clearTimeout` isn't necessary to keep the tests from shutting down as I'd thought.

Reply:
I'm trying to remove the unref on the timer.  I don't fully know the implications of calling unref on a timer in prod.

Reply:
I think it's not worth turning the timer on and off.  It's unnecessary complexity.  A 5 second timer that runs one if statement isn't much of a performance overhead.

Reply:
I think it's better off in `src`. It'd be unexpected to have a hanging process as a result of having executed that code path in a test. 

`setInterval` is intended to run continuously until `clearInterval` has been called, freeing the timer. If the rest of the process logic has completed, the process will remain running until any outstanding timers have completed. In our case there's no condition under which the timer will ever be stopped, which is why the `unref` is necessary.

A `SIGINT` **does** appear to stop the timer which is likely why we don't notice this while running the script locally. However, if you run the following line in your terminal you'll notice the process won't by itself:
```sh
$ node -e 'setInterval(() => {}, 1000)'
```

With `clearTimeout` (which ideally would be `clearInterval`, but the behavior _appears_ to be the same) in the tests, the additional `unref` would not be necessary. However, we can't always guarantee there's a future point in which we stop the timer unless we do so alongside it's instantiation.

Reply:
Did put `unref` in `src/`.  Used `clearInterval`.

Reply:
I've removed the need for all of this by:

a) Re-using (copying, unfortunately) the existing logic of EventAggregator, which uses `setTimeout` rather than `setInterval`
b) Making telemetry optional when the scanner is created - and disabling it in test cases